simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -0,0 +1,112 @@
+package io.prometheus.client.exporter;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.exporter.common.TextFormat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.URLDecoder;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.concurrent.Executors;
+
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.HttpExchange;
+
+/**
+ * Expose Prometheus metrics using a plain Java HttpServer.
+ * <p>
+ * Example Usage:
+ * <pre>
+ * {@code
+ * HTTPServer server = new HTTPServer(1234);
+ * }
+ * </pre>
+ * */
+public class HTTPServer {
+    static class HTTPMetricHandler implements HttpHandler {
+        private CollectorRegistry registry;
+
+        HTTPMetricHandler(CollectorRegistry registry) {
```

Comment:
Passing the registry is largely for unittesting, it's not something users should generally need.